### ISSUE TITLE
3D CNNs do not work

### ISSUE THREAD

**Issue by bulatGH** [Microsoft/CNTK#1573]
I have been trying to use 3D convolution networks.
I was selfvalidating the network on the training database. The results are ever numbers from 5-10, although the training labels are only 0 or 1, or the results are all zeros if I comment conv4_bn.
 

roughly the architecture is:

BrainScriptNetworkBuilder = [
        tsize=21
		tsize2=11
		imageShape = tsize:tsize:tsize:1
		lableShape = tsize2:tsize2:tsize2:1
		
		cMap1 = 20
		cMap2 = 20
		
		model = inputFeatures => {
                        conv1    = ConvolutionalLayer {cMap1, (4:4:4), pad = false, activation=ReLU}(inputFeatures)
			conv1_bn = BatchNormalizationLayer {spatialRank = 0} (conv1)
			conv1_d  = Dropout (conv1_bn)
			
			conv2    = ConvolutionalLayer {cMap1, (4:4:4), pad = false, activation=ReLU}(conv1_d)
			conv2_bn = BatchNormalizationLayer {spatialRank = 0} (conv2)
			conv2_d  = Dropout (conv2_bn)
			
			conv3    = ConvolutionalLayer {cMap1, (3:3:3), pad = false, activation=ReLU}(conv2_d)
			conv3_bn = BatchNormalizationLayer {spatialRank = 0} (conv3)
			conv3_d  = Dropout (conv3_bn)
			
			conv4    = ConvolutionalLayer {1, (3:3:3), pad = false, activation=ReLU}(conv3_d)
			conv4_bn = BatchNormalizationLayer {spatialRank = 0} (conv4)
			
        }.conv4
		
		# inputs
		features = Input {imageShape}
        labels   = Input {lableShape}

		#featScale = 1/2000
        #Scale{f} = x => Constant(f) .* x
		#f1 = Scale{featScale} (features)
		
		z = model (features)
		
		diff = z - labels
        sqErr = ElementTimes(diff, diff)
        mse = ReduceMean(sqErr)
        rmse = Sqrt(mse)
		
		zR = Round(z);
		tequalVal = Equal(zR, labels)
		errorVal = ReduceMean(tequalVal)
		
        # declare special nodes
        featureNodes    = (features)
        labelNodes      = (labels)
        criterionNodes  = (rmse)
        evaluationNodes = (errorVal)
        outputNodes     = (z)
    ]

**Comment by sayanpa** [comment]
Could you please confirm which version of the Beta release you are using? We have made some major changes recently and need to know your environment to be able to give you meaningful suggestions.

**Comment by bulatGH** [comment]
I am using the last version V 2.0 Beta 12.

**Comment by sayanpa** [comment]
We are looing into any possible regression in the operations. Assuming this is the first time you are trying 3D convolution on CNTK.

**Comment by sayanpa** [comment]
Could you please try out the CNTK V2 release and reopen the issue if it still not working for you. Also, could you try this with the python API's where significant effort has gone in the past quarter.
